### PR TITLE
Clean up string matcher output in cases where no single-character discriminator exists

### PR DESCRIPTION
**Pull request by adams85**

### CODE REVIEW

File: test/Esprima.Tests.SourceGenerators/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (lahma) [inline test/Esprima.Tests.SourceGenerators/Properties/launchSettings.json:1]: seems redundant?

File: src/Esprima.SourceGenerators/StringMatcherGenerator.cs [hunk 2]
```diff
@@ -131,14 +131,34 @@ private static void Execute(SourceProductionContext context, ImmutableArray<Stri
...
+            }
+
             var (typeModifiers, type) = typeGrouping.Key;
             sourceBuilder.Append(typeModifiers).Append(" partial class ").AppendLine(type)
                 .AppendLine("{");
 
+            bool firstMethod = true;
             foreach (var method in typeGrouping)
             {
+                if (firstMethod)
```
Comment (lahma) [inline src/Esprima.SourceGenerators/StringMatcherGenerator.cs:153]: nit: could just set to false after AppendLine

**Review by lahma — APPROVED** [review]
This is what I would call successful nerd-snipe, thanks for the cleanup!

File: test/Esprima.Tests.SourceGenerators/Properties/launchSettings.json (hunk 1 above)
Comment (adams85) [reply test/Esprima.Tests.SourceGenerators/Properties/launchSettings.json]: Yep, added accidentally.

**Comment by adams85** [comment]
Yeah, I managed to go full nerd with this indeed... 😄 But it felt good, so np. However, I'm running out of free time, so I can't promise pumping energy into the project with the same intensity as recently. I have [one more task](https://github.com/sebastienros/esprima-dotnet/issues/300) which I'd like to do for v3 if possible. How much would this interfere with your work? May I go ahead with this?

**Comment by lahma** [comment]
> Yeah, I managed to go full nerd with this indeed... 😄 But it felt good, so np. However, I'm running out of free time, so I can't promise pumping energy into the project with the same intensity as recently. 

And a big thank you for the latest sprint!

>I have [one more task](https://github.com/sebastienros/esprima-dotnet/issues/300) which I'd like to do for v3 if possible. How much would this interfere with your work? May I go ahead with this?

Fire at will, I'll manage the merge conflicts. You get things done which is more important than my forever-branches 😄

**Pull request merged**